typing/primitive.ml

```diff
@@ -20,11 +20,15 @@ open Parsetree
 
 type boxed_integer = Pnativeint | Pint32 | Pint64
 
+(* String, Bytes and Bigarray are not allowed as @untagged results *)
+type untagged_integer = Int | Bool | Char | Constants
+                        | String | Bytes | Bigarray
```

Comment:
`untagged_integer String` (or `Bytes`, or `Bigarray`) makes no sense to me. Could you restrict the scope of your PR to only cover actually immediate types, and discuss a bit more the design you have in mind for string/bytes/bigarray, separately?

Reply:
Thanks for the review! So quick! Yes sure I can split the PR in two, no problem. I will probably add Unix.file_descr, but these are not int on windows, so I should be carefull.

Reply:
For String, Bytes and Bigarray, what do you think of my proposal to use [[@direct]] (or a better name) that will cover both [[@untagged]], [[@unboxed]] and the new types ? Here I am not talking on implementation, just the look for the user.

Reply:
@gasche I redid the PR only for immediate type. Also updated a bit the top comment on this PR